wordpress/wp-content/plugins/memberful-wp/src/user/map.php

```diff
@@ -202,6 +204,22 @@ private function create_user() {
     $user_data['nickname']                = $this->member->full_name;
     $user_data['display_name']            = $this->member->full_name;
 
+    // Disable the `bkismet_check_signup` filter added by a platform plugin on
+    // wordpress.com.
+    if ( isset( $wp_filter['wp_pre_insert_user_data'] ) ) {
+      foreach ( $wp_filter['wp_pre_insert_user_data']->callbacks as $priority => $callbacks ) {
+        foreach ($callbacks as $callback_key => $callback) {
+          if ( is_array( $callback['function']) && is_object($callback['function'][0] ) ) {
+            $object = $callback['function'][0];
+
+            if ( method_exists( $object, 'bkismet_check_signup' ) ) {
+              remove_filter( 'wp_pre_insert_user_data', [$object, 'bkismet_check_signup'], $priority );
+            }
+          }
+        }
+      }
+    }
```

Comment:
Not blocking, but WDYT of extracting this to `core-ext.php` so we can share it with the rest of the plugin? I was thinking of something like:

`memberful_wp_remove_callback_from_filter('wp_pre_insert_user_data', 'bkismet_check_signup')`

Or will we run into trouble due to having to pass `$object` to some callbacks but not to others? I'm not familiar with what's inside `$callback`, so I don't know how simple this would be. I'll defer to you.

Reply:
Good question. In general, it should be enough to call [remove_filter()](https://developer.wordpress.org/reference/functions/remove_filter/) directly because we know what `$callback` is.

`$callback` can be a function name, an array containing a class instance or a class name with a method, or an anonymous function. In this case, it is a class instance with a method name. However, since we don’t know the class instance, we need to iterate over all callbacks and remove the one where the method name matches.

I think this is a special case and not something we should be doing regularly, so I prefer keeping this code here rather than extracting it into a function.

Reply:
Makes sense, thanks!